internals/secrethub/tpl/v2.go

```diff
@@ -188,8 +188,15 @@ func (p *v2Parser) parseRoot() (node, error) {
 		return variable, p.readRune()
 	}
 
-	if p.current == token.Dollar && (unicode.IsLetter(p.next) || p.next == '_') {
-		return nil, ErrUnexpectedDollar(p.lineNo, p.columnNo)
+	if p.current == token.Dollar && p.isVariableStartRune(p.next) {
+		variable, err := p.parseVarWithoutBrackets()
+		if err == io.EOF {
```

Comment:
Do we need this check here?

Reply:
We could also just `return p.parseVarWithoutBrackets()`

Reply:
My thought exactly. Keeping it short; keeping it simple :smile: